src/compiler/parser.ts

```diff
@@ -4968,14 +5113,15 @@ module ts {
 
         function parseDeclaration(): ModuleElement {
             let fullStart = getNodePos();
+            let decorators = parseDecorators();

```

Comment:
This allows you to use decorators *before* the `export` keyword.
```javascript
@decorator export default class Foo {}
```
But if [this](https://github.com/wycats/javascript-decorators) is the spec we're following, this isn't actually included in the grammar. Only:

```javascript
export default @decorator class Foo {}
```